src/main/js/App.js

```diff
@@ -1,11 +1,259 @@
+                    ...questions.slice(0, i),
+                    {
+                        type: currentType,
+                        min: 0,
+                        max: 0,
+                        question: '',
+                    },
+                    ...questions.slice(i, questions.length)
+                ]);
+                break;
+            default:
+                console.log(`[WARNING] Unknown question type "${currentType}"`);
+        }
+    };
+
+    /*
+    * A helper function to update the array of
+    * current questions when a change is made
+    */
+    const updateQuestion = (i, newObjVal) => {
+        setQuestions(questions.map((q, current) => {
+            if (current === i) {
+                return newObjVal;
+            }
+            return q;
+        }));
+    };
+
+    /*
+    * Delete a question at index i
+    */
+    const deleteQuestion = (i) => {
+        setQuestions(questions.filter((q, current) => {
+            if (current === i) {
+                return false;
+            }
+            return true;
+        }));
+    };
+
+    /*
+    * Store a new survey on the server
+    */
+    const createSurvey = () => {
+        const survey = {
+            name: surveyName,
+            questions,
+        };
+
+        fetch(`${webUrl}createSurvey`, {
+            method: 'POST',
+            body: JSON.stringify(survey),
+            headers: {
+                'Content-Type': 'application/json'
+            },
+        })
+        .then(res => res.json())
+        .then(display)
+        .catch(display);
+    };
+
+    return (
+        <div className="qq-app">
+            <Typography variant="h4" gutterBottom>Question Quail</Typography>
+            <div>
+                <TextField
+                    className="qq-app mv"
+                    variant="outlined"
+                    label="Survey Name"
+                    size="small"
+                    onChange={e => setSurveyName(e.target.value)}
+                />
+                <Button className="qq-app m" variant="contained" color="primary" onClick={createSurvey}>Create</Button>
+            </div>
+            <div>
+                <FormControl className="qq-app mv qq-app__qtype_select">
+                    <InputLabel id="qtype_select_label">Question Type</InputLabel>
+                    <Select labelId="qtype_select_label" value={currentType} onChange={e => setCurrentType(e.target.value)}>
+                        <MenuItem value={qType.OPEN_ENDED}>Open-Ended</MenuItem>
+                        <MenuItem value={qType.NUMERICAL}>Numerical</MenuItem>
+                    </Select>
+                </FormControl>
+                <Button className="qq-app m" variant="contained" color="primary" disabled={!currentType} onClick={addQuestion}>+</Button>
+            </div>
+            <div>
+                <Button className="qq-app mv" variant="contained" color="primary" onClick={myFunc1}>Dev Test 1 (POST)</Button>
```

Comment:
oh no "myFunc1" has persisted from the other PR...

Reply:
The discussion ended with us leaving the naming intact for the time being, as this is a developer test function.